Source/Scene/Vector3DTileContent.js

```diff
@@ -0,0 +1,324 @@
+            requestFunction : loadArrayBuffer,
+            type : RequestType.TILES3D,
+            distance : distance
+        }));
+        if (defined(promise)) {
+            this.state = Cesium3DTileContentState.LOADING;
+            promise.then(function(arrayBuffer) {
+                if (that.isDestroyed()) {
+                    return when.reject('tileset is destroyed');
+                }
+                that.initialize(arrayBuffer);
+            }).otherwise(function(error) {
+                that.state = Cesium3DTileContentState.FAILED;
+                that.readyPromise.reject(error);
+            });
+        }
+    };
+
+    function createColorChangedCallback(content) {
+        return function(batchId, color) {
+            if (defined(content._primitive)) {
+                content._primitive.updateCommands(batchId, color);
+            }
+        };
+    }
+
+    var sizeOfUint16 = Uint16Array.BYTES_PER_ELEMENT;
+    var sizeOfUint32 = Uint32Array.BYTES_PER_ELEMENT;
+
+    /**
+     * Part of the {@link Cesium3DTileContent} interface.
+     */
+    Vector3DTileContent.prototype.initialize = function(arrayBuffer, byteOffset) {
+        byteOffset = defaultValue(byteOffset, 0);
+
+        var uint8Array = new Uint8Array(arrayBuffer);
+        var magic = getMagic(uint8Array, byteOffset);
+        if (magic !== 'vctr') {
+            throw new DeveloperError('Invalid Vector tile.  Expected magic=vctr.  Read magic=' + magic);
+        }
+
+        var view = new DataView(arrayBuffer);
+        byteOffset += sizeOfUint32;  // Skip magic number
+
+        //>>includeStart('debug', pragmas.debug);
+        var version = view.getUint32(byteOffset, true);
+        if (version !== 1) {
+            throw new DeveloperError('Only Vector tile version 1 is supported.  Version ' + version + ' is not.');
+        }
+        //>>includeEnd('debug');
+        byteOffset += sizeOfUint32;
+
+        var byteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        if (byteLength === 0) {
+            this.state = Cesium3DTileContentState.PROCESSING;
+            this.contentReadyToProcessPromise.resolve(this);
+            this.state = Cesium3DTileContentState.READY;
+            this.readyPromise.resolve(this);
+            return;
+        }
+
+        var featureTableJSONByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var featureTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var batchTableJSONByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var batchTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var indicesByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var positionByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
```

Comment:
+1.  The trend has been to use the binary header largely as just metadata for the tile, and then push feature-specific info into the feature table.